habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomMonorepoGroupMigration.java

```diff
@@ -0,0 +1,147 @@
+ * project's README.md, this prevents these dependencies from causing issues when Poetry projects are exported in
+ * development releases.
+ */
+public class CustomMonorepoGroupMigration extends AbstractMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(CustomMonorepoGroupMigration.class);
+
+    protected Map<String, TomlReplacementTuple> replacements = new HashMap<>();
+
+    protected boolean hasExistingMonoRepoDependenciesGroup;
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        replacements.clear();
+        hasExistingMonoRepoDependenciesGroup = false;
+
+        boolean shouldExecute = false;
+        try (FileConfig tomlFileConfig = FileConfig.of(file)) {
+            tomlFileConfig.load();
+
+            Optional<Config> toolPoetryDependencies = tomlFileConfig.getOptional(TomlUtils.TOOL_POETRY_DEPENDENCIES);
+            if (toolPoetryDependencies.isPresent()) {
+                Config foundDependencies = toolPoetryDependencies.get();
+                Map<String, Object> dependencyMap = foundDependencies.valueMap();
+
+                for (Map.Entry<String, Object> dependency : dependencyMap.entrySet()) {
+                    String packageName = dependency.getKey();
+                    Object packageRhs = dependency.getValue();
+                    if (TomlUtils.representsLocalDevelopmentVersion(packageRhs)) {
+                        String packageRshAsString = TomlUtils.convertCommentedConfigToToml((CommentedConfig) packageRhs);
+                        logger.info("Found local dependency not within monorepo group! ({} = {})", packageName, packageRshAsString);
+                        TomlReplacementTuple replacementTuple = new TomlReplacementTuple(packageName, packageRshAsString, "");
+                        replacements.put(packageName, replacementTuple);
+                        shouldExecute = true;
+                    }
+                }
+            }
+
+            Optional<Config> toolPoetryMonorepoDependencies = tomlFileConfig.getOptional(TomlUtils.TOOL_POETRY_GROUP_MONOREPO_DEPENDENCIES);
+            if (toolPoetryMonorepoDependencies.isPresent()) {
+                hasExistingMonoRepoDependenciesGroup = true;
+            }
+        }
+
+        return shouldExecute;
+    }
+
+    @Override
+    protected boolean performMigration(File pyProjectTomlFile) {
+        String fileContent = StringUtils.EMPTY;
+        try (BufferedReader reader = new BufferedReader(new FileReader(pyProjectTomlFile))) {
+            String line = reader.readLine();
+            boolean injectAfterNextEmptyLine = false;
+
+            while (line != null) {
+                boolean addLine = true;
+                boolean isEmptyLine = line.isBlank();
+
+                if (line.contains(StringUtils.SPACE) || line.contains(TomlUtils.EQUALS)) {
+                    String key = line.substring(0, line.indexOf(StringUtils.SPACE));
```

Comment:
S: Somewhat nervous of exceptions in the case where `line.contains(StringUtils.SPACE) == false`.  I want to say (could be very wrong) that `indexOf` will return `-1`, and then `substring(0, -1)` throws an exception.

Reply:
I could also be completely wrong here, but I feel like I remember substring not liking that.

Reply:
✅ Resolved